### DISCUSSION TITLE
How to build MSEide in another language (using mseconsts_*)

### DISCUSSION THREAD

**Discussion by rchastain** [mse-org/mseide-msegui#81] [Q&A]
Hello everybody! Who knows how to build MSEide in another language? How do we use _mseconsts_*_ files?

**Comment by rchastain** [comment]
It looks like I found the answer (not yet tested).

```
In the main program add mselangconst and mselangconst_ru to uses, call
 setlangconsts('ru');
```

**Reply by fredvs** [reply]
Hello Roland.

This should be a very old note, before release 4.6.2.
In 4.6.2 _mselangconst.pas_ was renamed to _mseconst.pas_ and all  _mselangconst_xx.pas_ into _mseconst_xx.pas._

https://gitlab.com/mseide-msegui/mseide-msegui/-/tree/master/lib/common/kernel?ref_type=heads

Note too that doing this will only translate the constants used by (some of) the widgets, not the captions of _MSEide_ himself, like menu-caption, buttons-caption, etc.

If you want a complete translated _MSEide_, this is lot of work, already tried using msei18n with Alexey 5 years ago but I give up.

Anyway, if you want a _MSEide_-fork completely translated in many languages, using dynamic loaded po files, you may try [ideU](https://github.com/fredvs/ideU). ;-)

**Reply by rchastain** [reply]
Thank you Fred. Yes, I saw that ideU is multilanguage. For me it is in French, without doing nothing. How is the user language detected?

**Comment by fredvs** [comment]
Hello Roland.

>How is the user language detected?

 `Gettext.GetLanguageIDs(MSELang, MSEFallbackLang);`


With ideU, you may changing-set the language with the "world" icon on the main menu and also define the font to use for each language.
